src/custom/utils/misc.ts

```diff
@@ -6,6 +6,20 @@ export const isTruthy = <T>(value: T | null | undefined | false): value is T =>
 export const delay = <T = void>(ms = 100, result?: T): Promise<T> =>
   new Promise(resolve => setTimeout(resolve, ms, result))
 
+export function isPromiseFulfilled<T>(
+  promiseResult: PromiseSettledResult<T>
+): promiseResult is PromiseFulfilledResult<T> {
+  return promiseResult.status === 'fulfilled'
+}
+
+// To properly handle PromiseSettleResult which returns and object
```

Comment:
I feel the comment should say what if does, not why you did it. 

sth like `Returns either a fullfilled value or a default`

actually, even the name can hint the responsability `getPrimiseFullfilledOrDefault`

Reply:
noted in https://github.com/gnosis/cowswap/issues/751